lib/cli-update-connection.ts

```diff
@@ -61,26 +69,26 @@ export default function CliUpdateConnection() {
             connectionString: settings.connectionString
         };
 
+        cli.print('Updating connection string...');
         client.datasets.setAllConnections(settings.collection, settings.workspace, settings.dataset, params, (err, result) => {
-            if (err) {
-                cli.error('Error updating connection string');
-                cli.error(err);
-                return;
+            if (err && typeof (callback) === 'function') {
+                callback(err);
             }
 
             cli.success('Connection string successfully updated');
             cli.print('Dataset: ', settings.dataset);
             cli.print('ConnectionString: ', settings.connectionString);
 
-            if (callback) {
+            if (result && typeof (callback) === 'function') {
                 callback(null, result);
             }
         });
     }
 
     function updateCredentials(client: powerbi.PowerBIClient, settings: any, callback?): void {
+        cli.print('Getting gateway datasources...');
         client.datasets.getGatewayDatasources(settings.collection, settings.workspace, settings.dataset, (err, result) => {
-            if (err) {
+            if (err && typeof (callback) === 'function') {
                 return callback(err);
```

Comment:
These are private funcs and we now always pass a callback.  It should be fine to change to required params instead of optional as it is now.